pkg/directory/setup.go

```diff
@@ -37,7 +37,7 @@ func Setup(parent *mux.Router, db *mongo.Database) error {
 	userAuthenticated.Use(userAuthMW.Middleware)
 	nodesAuthenticated.Use(nodeAuthMW.Middleware)
 
-	nodes.HandleFunc("", mw.AsHandlerFunc(nodeAPI.registerNode)).Methods("POST").Name("node-register")
+	nodesAuthenticated.HandleFunc("", mw.AsHandlerFunc(nodeAPI.registerNode)).Methods("POST").Name("node-register")
```

Comment:
You cannot really it like this, cause the first time a node is booted, it's public key is not yet registered. So this authentication middleware will returns an error since the node doesn't not exists yet.